gammapy/utils/serialization/io.py

```diff
@@ -246,11 +216,12 @@ def add_background(self, dataset, component, bkg_prev):
             try:
                 cube = self.cube_register[component["name"]]
             except KeyError:
+                print(component)
                 cube = SkyDiffuseCube.read(component["filename"])
```

Comment:
how do you know the filename used to create the background model when you don't use from_skymodel.() ?
you have to specify the filename in the call and serialise it in .to_dict() if not None, right ?